src/common/http/post_lambda_request.cpp

```diff
@@ -0,0 +1,31 @@
+#include "post_lambda_request.hpp"
+
+namespace http {
+PostLambda::PostLambda(LambdaT lam, const char *url_string)
+    : hdrs {
+        { nullptr, nullptr, std::nullopt },
+    }
+    , url_string(url_string)
+    , lamb(lam) {
+}
+const char *PostLambda::url() const {
+    return url_string;
+}
+ContentType PostLambda::content_type() const {
+    return ContentType::ApplicationOctetStream;
+}
+Method PostLambda::method() const {
+    return Method::Post;
+}
+const HeaderOut *PostLambda::extra_headers() const {
+    return hdrs;
+}
+std::variant<size_t, Error> PostLambda::write_body_chunk(char *data, size_t size) {
+    if (!size || !lamb) {
+        return Error::InternalError;
+    }
```

Comment:
I think you'd prefer to assert on these things, because they are likely bugs, not errors to expect and somehow deal with at runtime.

Reply:
I agree partially, I will add the assert but I think the check should still be there (better to 'not do anything' than 'crash when invoking a nullptr')